frontend/src/actions/TableActions.js

```diff
@@ -511,8 +520,12 @@ function updateCollapsedRows({ tableId, rows }) {
       collapsedRows,
       collapsedParentRows,
     } = table;
-    let newCollapsedParentRows = [];
-    let newCollapsedRows = [];
+
+    const hasChangedIds = changedIds && changedIds.length > 0;
+    let newCollapsedParentRows =
+      hasChangedIds || !uncollapseRowsOnChange ? [...collapsedParentRows] : [];
+    let newCollapsedRows =
+      hasChangedIds || !uncollapseRowsOnChange ? [...collapsedRows] : [];
```

Comment:
i think all these lines shall be moved down, 
inside those 2 ifs.....


because if it's not collapsible and or rows is empty then your computation is not neeed.

Reply:
Inside first if. would make sense `if (collapsible) {`

If you move inside the 2nd one `if (rows.length) {`  you will not have it defined for the dispatch